src/components/policy/policy_external/src/policy_helper.cc

```diff
@@ -620,33 +620,50 @@ void FillNotificationData::UpdateParameters(
   ParametersConstItr it_parameters = in_parameters.begin();
   ParametersConstItr it_parameters_end = in_parameters.end();
 
-  // Due to APPLINK-24201 SDL must consider cases when 'parameters' section is
-  // not present for RPC or present, but is empty.
-
-  // If 'parameters' section is like: 'parameters' : []
-  if (in_parameters.is_initialized() && in_parameters.empty()) {
-    if (!does_require_user_consent_) {
-      out_parameter.any_parameter_disallowed_by_policy = true;
-    }
-    if (does_require_user_consent_ && kAllowedKey == current_key_) {
-      out_parameter.any_parameter_disallowed_by_user = true;
+  // From AppLink Policies Manager specification:
+  // To determine consent for a particular RPC in a particular HMI level with
+  // particular parameters (if applicable), the system shall find all of the
+  // functional groups the RPC is included in. If user consent is needed as
+  // listed within the functional group in the policy table, the system shall
+  // use a logical AND: backend permissions AND User permissions. If the RPC is
+  // listed under more than one group, the system shall perform a logical OR
+  // amongst all of the possible allowed permissions scenarios for the RPC (and
+  // parameter/or HMI level) defined by each of the functional groups.
+
+  if (!IsSomeParametersAllowed(out_parameter)) {
+    // Due to APPLINK-24201 SDL must consider cases when 'parameters' section is
+    // not present for RPC or present, but is empty.
+
+    // If 'parameters' section is like: 'parameters' : []
+    if (in_parameters.is_initialized() && in_parameters.empty()) {
+      if (!does_require_user_consent_) {
+        out_parameter.any_parameter_disallowed_by_policy = true;
+      }
+      if (does_require_user_consent_ && kAllowedKey == current_key_) {
+        out_parameter.any_parameter_disallowed_by_user = true;
+      }
     }
-  }
 
-  // If 'parameters' section is omitted
-  if (!in_parameters.is_initialized()) {
-    if (!does_require_user_consent_) {
-      out_parameter.any_parameter_allowed = true;
-    }
-    if (does_require_user_consent_ && kAllowedKey == current_key_) {
-      out_parameter.any_parameter_allowed = true;
+    // If 'parameters' section is omitted
+    if (!in_parameters.is_initialized()) {
+      if (!does_require_user_consent_ ||
+          (does_require_user_consent_ && kAllowedKey == current_key_)) {
+        out_parameter.any_parameter_allowed = true;
```

Comment:
I don't think this gets set if `IsSomeParametersAllowed()` returns true at the top conditional

Reply:
This part should be outside of the line 633 conditional I believe

Reply:
@conlain-k good catch. I removed outer conditional and now it works as expected. See 80291097033c8a6cd44a95f2ff4ff4def57e5e8e